python/python/lancedb/pydantic.py

```diff
@@ -37,7 +37,7 @@
 import pydantic
 import semver
 
-PYDANTIC_VERSION = semver.parse_version_info(pydantic.__version__)
+PYDANTIC_VERSION = semver.Version.parse(pydantic.__version__)
```

Comment:
Yeah this function was introduced in `semver` version 3.0.0, so using this breaks compatibility with 2.x versions of `semver`.

One alternative I've suggested is using `packaging.version.parse()` from [packaging](). This is a more common dependency than `semvar`, so it's likely users already have it downloaded, and the parse function there has been around since nearly the earliest version of it and supports parsing semver versions.

Reply:
Happy to change the PR to depend on packaging instead. And I agree, that is indeed the much more common dependency.

Reply:
It might be a little more work, since the version object returned isn't quite the same. But that seems like the best solution to me.